src/Query/DateParser.ts

```diff
@@ -20,36 +20,54 @@ export class DateParser {
      * @return - A Tuple of dates. If both input dates are invalid, then both ouput dates will be invalid.
      */
     public static parseDateRange(input: string): [moment.Moment, moment.Moment] {
+        // Try parsing a relative date range like 'current month'
         const relativeDateRange = DateParser.parseRelativeDateRange(input);
         if (relativeDateRange !== undefined) {
             return relativeDateRange;
         }
 
-        // If relative date range was not parsed, fallback on absolute date range with chrono
-        return DateParser.parseAbsoluteDateRange(input);
+        // Try '2022-W10' otherwise
+        const specificDateRange = DateParser.parseSpecificDateRange(input);
+        if (specificDateRange !== undefined) {
+            return specificDateRange;
+        }
+
+        // If previous failed, fallback on absolute date range with chrono
+        const absoluteDateRange = DateParser.parseAbsoluteDateRange(input);
+        if (absoluteDateRange !== undefined) {
+            return absoluteDateRange;
+        }
```

Comment:
When I see patterns like this, I look for ways to simplify them.

I don't know the exact syntax, but I bet there is a way to say:

```ts
const converters = [
    DateParser.parseRelativeDateRange,
    DateParser.parseSpecificDateRange,
    DateParser.parseAbsoluteDateRange ];
```

Then loop over the converters, calling them, and returning their result if it is defined.

Reply:
(The pattern being 3 lines of code that differs in only one thing, which here is the function being called)

Reply:
ah, that's a nice way to put this, I will take the pattern from `FilterInstructions.createFilterOrErrorMessage()